chia/wallet/wallet_state_manager.py

```diff
@@ -686,10 +707,75 @@ async def determine_coin_type(
+        coin_spends: List[CoinSpend] = []
+        tx_fee = uint64(self.config.get("auto_claim_tx_fee", 0))
+        for coin in unspent_coins:
+            metadata = json.loads(coin.metadata)
+            if coin_state.coin.name() == coin.coin.name() and coin_state.spent_height is not None:
+                # This coin is spent, update it
+                self.log.info(f"Marked merkle coin {coin.coin.name().hex()} as spent")
+                await self.merkle_coin_store.set_spent(coin.coin.name(), uint32(coin_state.spent_height))
+                continue
+            if coin_state.coin.parent_coin_info == coin.coin.name() and coin_state.created_height is not None:
+                # This coin is spent, update it
+                self.log.info(f"Marked merkle coin {coin.coin.name().hex()} as spent")
+                await self.merkle_coin_store.set_spent(coin.coin.name(), uint32(coin_state.created_height))
+                continue
+            if (
+                self.config.get("auto_claim", True)
+                and self.config.get("auto_claim_min_amount", 0) < coin.coin.amount
+                and coin.coin_type == MerkleCoinType.CLAWBACK.value
+                and metadata["is_recipient"]
+                and coin.spent == 0
+            ):
+                coin_timestamp = await self.wallet_node.get_timestamp_for_height(coin.confirmed_block_height)
+                if current_timestamp - coin_timestamp >= metadata["time_lock"]:
+                    recipient_puzhash = bytes32.from_hexstr(metadata["recipient_puzhash"])
+                    derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(recipient_puzhash)
+                    assert derivation_record is not None
+                    inner_puzzle: Program = self.main_wallet.puzzle_for_pk(derivation_record.pubkey)
+                    inner_solution: Program = self.main_wallet.make_solution(
+                        primaries=[{"puzzlehash": recipient_puzhash, "amount": uint64(coin.coin.amount), "memos": []}],
+                        coin_announcements=None if len(coin_spends) > 0 or tx_fee == 0 else {coin.coin.name()},
+                    )
+                    coin_spends.append(
+                        generate_clawback_spend_bundle(coin.coin, metadata, inner_puzzle, inner_solution)
+                    )
+                    if len(coin_spends) >= self.config.get("auto_claim_coin_size", 50):
+                        await self.claim_clawback_coins(coin_spends, tx_fee)
+                        coin_spends = []
+        if len(coin_spends) > 0:
+            await self.claim_clawback_coins(coin_spends, tx_fee)
+
+    async def claim_clawback_coins(self, coin_spends: List[CoinSpend], fee: uint64):
+        assert len(coin_spends) > 0
+        spend_bundle = await self.main_wallet.sign_transaction(coin_spends)
+        if fee > 0:
+            announcement_to_make = coin_spends[0].coin.name()
```

Comment:
You should use like a sorted hash of all coin IDs for this entropy instead.  Its safer if the entire transaction is secured that way.

Reply:
Fixed